target/linux/generic/files/drivers/net/phy/rtl8367b.c

```diff
@@ -1560,32 +1552,50 @@ static int rtl8367b_detect(struct rtl8366_smi *smi)
 
 	switch (chip_ver) {
 	case 0x0020:
-		if (chip_num == 0x6367)
+		if (chip_num == 0x6367) {
 			chip_name = "8367RB-VB";
+			smi->rtl8367b_chip = RTL8367B_CHIP_RTL8367RB_VB;
+		}
 		break;
 	case 0x00A0:
-		if (chip_num == 0x6367)
+		if (chip_num == 0x6367) {
 			chip_name = "8367S";
+			smi->rtl8367b_chip = RTL8367B_CHIP_RTL8367S;
+		}
 		break;
 	case 0x1000:
 		chip_name = "8367RB";
+		smi->rtl8367b_chip = RTL8367B_CHIP_RTL8367RB;
 		break;
 	case 0x1010:
 		chip_name = "8367R-VB";
+		smi->rtl8367b_chip = RTL8367B_CHIP_RTL8367R_VB;
 	}
 
 	if (!chip_name) {
+		rlvid = (chip_ver >> RTL8367B_CHIP_VER_RLVID_SHIFT) &
+			RTL8367B_CHIP_VER_RLVID_MASK;
```

Comment:
The driver will always stop probing when it reaches this code. 
The `rtl8367b_init_regs` function makes use of some enum values only st here.

The old code in `rtl8367b_init_regs` would also trigger for `RTL8367B_CHIP_RTL8367RB` and `RTL8367B_CHIP_RTL8367R_VB`. Are you sure this is not used to distinguish them more detailed?

Reply:
This code only runs when !chip_name. RTL8367B_CHIP_RTL8367RB and RTL8367B_CHIP_RTL8367R_VB is done in the earlier code (a few lines above).

Reply:
This code only runs when !chip_name. Detection of RTL8367B_CHIP_RTL8367RB and RTL8367B_CHIP_RTL8367R_VB is done in the earlier code (a few lines above). Code was tested for RTL8367B_CHIP_RTL8367S,
Assignments are the some as in https://github.com/openwrt/openwrt/pull/15033#issue-2217119665

Reply:
In the new code it was only executed for unsupported chips, in the old code it was also executed for some supported chips, see:
https://github.com/openwrt/openwrt/pull/15749/files?diff=split&w=0#r1693980451

Reply:
I looked a bit closer at the code and it should be fine if you remove the code for `rtl8367r_vb_initvals_0` and the code setting `RTL8367B_CHIP_UNKNOWN_RLVID0` and `RTL8367B_CHIP_UNKNOWN_RLVID1` as they did not work before too.

I overlooked before that the `rlvid` was calculated based on the `chip_ver`.